Feat(PP-63):added detailed results game description

DmitrijSidorkin opened the pull request:
Created a new model for storing detailed game data that is later being fetched in `detailed-results` page and is displayed there.

elnino97 commented on public/javascripts/detailedResultsSlides.js:
```diff
@@ -1,5 +1,61 @@
+        <div class="game-details-item">Platforms:<p class="details-text">${
+          data.platforms
+        }</p></div>
+        <div class="game-details-item">Stores:${stores}</div>
```
If no store or website maybe you could not render these elements at all:

![image](https://github.com/DmitrijSidorkin/pixel-thing/assets/9388169/23a12a4a-f7d6-4e8b-a91d-f06af81ddb5a)

elnino97 commented on public/javascripts/detailedResultsSlides.js:
```diff
@@ -1,5 +1,61 @@
+      if (data.metaScore >= 75) {
+        metaScoreStyle = "meta-green";
+      } else if (data.metaScore >= 50) {
+        metaScoreStyle = "meta-yellow";
```
What you're defining here is not style I guess so let's change the name to class instead

elnino97 commented on public/stylesheets/detailed-results.css:
```diff
@@ -104,27 +104,81 @@ body {
+  display: flex;
+  flex-direction: column;
+  align-self: end;
+  margin: 0px 30px 20px 0px;
```
I think I mentioned it long time ago but you should a padding on the card wrapper rather than giving each element their separate spacing styles. If you do that, you'll fix another issue which is the fact that spacing is not unified across the card:
![image](https://github.com/DmitrijSidorkin/pixel-thing/assets/9388169/64300f33-0ad0-4125-8a0b-f672b95bdd08)

DmitrijSidorkin replied:
Did try to change where i put the margins a bit, this specific line for meta score did not really change though since that placement was kinda intentional, unless i am missing something here.

elnino97 replied:
Well the spacing is all over the place imo, the text is touching the side of the card, like overall there should be unified padding in the card in general. If you look at the meta score element then you see it has its own padding to the right (why tho)

DmitrijSidorkin replied:
Felt like meta-score being pushed a bit to the right was visually more appealing :d

elnino97 replied:
I agree about meta data, but the text touching the card border is a bit eh. At the end of the day we don't really have a design so kinda have to eyeball everything, and so this was my general impression of how the page looks.

elnino97 commented on public/stylesheets/detailed-results.css:
```diff
@@ -104,27 +104,81 @@ body {
-  background-color: #333;
+  overflow-wrap: break-word;
+}
+.game-details-item {
```
Let's try to write descriptive class names, rather than `item`, how about `text-wrapper`?

elnino97 commented on public/stylesheets/detailed-results.css:
```diff
@@ -104,27 +104,81 @@ body {
+  font-size: 20px;
+  margin-bottom: 15px;
+}
+.details-text,
```
This is a child element of `game-details-item` so it would be preferable if the naming followed the structure as well, per say: `game-details-text`?

elnino97 commented on public/javascripts/detailedResultsSlides.js:
```diff
@@ -1,5 +1,61 @@
+          data.developers
+        }</p></div>
+        <div class="game-details-item">Platforms:<p class="details-text">${
+          data.platforms
```
I think platforms are missing spaces " " after each ","

DmitrijSidorkin replied:
Same thing was going on with genres and devs

elnino97 commented on middleware/middleware.js:
```diff
@@ -60,6 +61,50 @@ module.exports.fetchRandomGameData = async (req, res, next) => {
 };
 
+//function for making API requests for additional data
+module.exports.fetchAdditionalGameData = async (fetchLink) => {
```
```suggestion
module.exports.fetchAdditionalGameData = async (fetchLink) => {
  try {
    const response = await axios.get(fetchLink);
    return response.data;
  } catch (error) {
    if (process.env.IN_DEVELOPMENT === "true") {
      console.log(error.response.status);
    }
  }
};
```

elnino97 commented on middleware/middleware.js:
```diff
@@ -60,6 +61,50 @@ module.exports.fetchRandomGameData = async (req, res, next) => {
+};
+
+//adds more detailed game data (stores, game website and developers) and stores in database
+module.exports.completeAndStoreGameData = async (detailedGameData) => {
```
I would rewrite this segment like this to make it a bit more structured and easier to read:

```suggestion
module.exports.completeAndStoreGameData = async (detailedGameData) => {
  //making API requests to fetch developers, website and store links for the game
  const GAME_DATA_URL = `${apiUrl}games/${detailedGameData.gameId}?key=${process.env.RAWG_KEY}`;
  const STORES_URL = `${apiUrl}games/${detailedGameData.gameId}/stores?key=${process.env.RAWG_KEY}`;

  const gameData = await this.fetchAdditionalGameData(GAME_DATA_URL);
  const gameStoresResponse = await this.fetchAdditionalGameData(STORES_URL);

  const developerNames = gameData.developers.map((developer) => developer.name);
  const gameStoresLinks = gameStoresResponse.results.map(
    ({ url, store_id }) => ({ url, store_id })
  );

  const updatedGameData = {
    ...detailedGameData,
    website: gameData.website,
    developers: developerNames,
    stores: gameStoresLinks,
  };

  //saving data in mongodb
  const newGameData = new GameData({
    ...updatedGameData,
  });
  await newGameData.save();
};

```

elnino97 commented on middleware/middleware.js:
```diff
@@ -70,9 +115,32 @@ module.exports.fetchRandomGameDataArr = async (req, res) => {
 
   if (response.status === 200) {
+    //add a conditional to see if the game data is already in mongodb
+    const alreadyExists = await GameData.findOne({
```
Not sure about this name, data returned is an object?

DmitrijSidorkin replied:
I guess the naming could be better, but it should either be empty or return a game object.

DmitrijSidorkin commented:
Changed some styling, changed javascript slides generation a bit so that empty elements are not displayed anymore and there is proper spacing between each genre/platform/developer. The suggested changed to middleware functions got added as well, though when applied in github, instead of replacing the former functions they were duplicated? Fixed that as well.
